imports/lib/schemas/ChatNotification.ts

```diff
@@ -1,72 +1,28 @@
-import * as t from 'io-ts';
-import { date } from 'io-ts-types';
-import { BaseCodec, BaseOverrides } from './Base';
+import { z } from 'zod';
 import { ChatMessageContent } from './ChatMessage';
+import { nonEmptyString } from './customTypes';
 import { Id } from './regexes';
-import type { Overrides } from './typedSchemas';
-import { buildSchema, inheritSchema } from './typedSchemas';
+import withCommon from './withCommon';
 
-const SharedFields = {
+// A notification triggered by a chat message sent by a user.
+const ChatNotification = withCommon(z.object({
   // The userId of the user for whom this notification targets.
-  user: t.string,
+  user: z.string().regex(Id),
```

Comment:
I notice we use this `z.string().regex(Id)` a bunch for all the foreign-key references.  Do you have an opinion on whether it would be good or counterproductive to have a `customType` for an id from another collection?  I could see it being neat to be able to potentially traverse referenced objects, but I also don't see it being super likely we write any bugs where we pull from the wrong model in the first place, and if we named the collection the foreign key referenced then we'd have schemas depending on model types, which seems kinda messy (we'd still have a DAG, but it'd be hairier).

I think my position is: writing this as `foreignKey` could be a neat shorthand to have (where `foreignKey` is just `z.string().regex(Id)`, but we should punt for now on referencing other models, and also this is fine as written.

Reply:
I actually kind of want to play with using [branded types](https://zod.dev/?id=brand) for foreign keys, which would allow us to capture at a type level what type of foreign key we're interacting with. (And potentially make it a type error to, e.g., pass a `Puzzle` ID when we're expecting a `Hunt` ID, or whatever)

But yeah, I think it makes sense to have a `foreignKey` helper for the time being. I'll add that.